lib/ui/login/login_view_model.dart

```diff
@@ -0,0 +1,62 @@
+import 'package:flutter_survey/model/login_model.dart';
+import 'package:flutter_survey/usecases/base/base_use_case.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_survey/ui/login/login_state.dart';
+import 'package:flutter_survey/usecases/log_in_use_case.dart';
+import 'package:flutter_survey/utils/storage/secure_storage.dart';
+import 'package:rxdart/subjects.dart';
+import 'package:email_validator/email_validator.dart';
+
+class LoginViewModel extends StateNotifier<LoginState> {
+  final LogInUseCase _loginUseCase;
+  final SecureStorage _secureStorage;
```

Comment:
The VM should not interact directly with a Storage or Repository. The action to save the data could be moved to the `LogInUseCase`, so `LogInUseCase` will include `SecureStorage` instead.

Reply:
Yes agree, let's move `_secureStorage` to `LogInUseCase` 🙏